### PR TITLE
Mention tool lint attributes

### PR DESCRIPTION
**Pull request by Manishearth**
Being stabilized in https://github.com/rust-lang/rust/pull/54870 , wanted to preserve some documentation. Clippy has more docs on this.

r? @steveklabnik

### CODE REVIEW

File: src/attributes.md [hunk 1]
```diff
@@ -357,6 +357,35 @@ pub mod m3 {
 }
 ```
 
+#### Tool lint attributes (`#[allow(clippy::foo)]`)
+
+Tool lints let you use scoped lints, to `allow`, `warn`, `deny` or `forbid` lints of
+certain tools.
+
+
```
Comment (matthewjasper) [inline src/attributes.md]: extra new line

Comment (Manishearth) [reply src/attributes.md]: fixed

**Comment by Havvy** [comment]
This is okay as is. Don't take the rest of what I'm saying as a requirement to make changes, although do tell me if you want to make them. Otherwise, we can merge this as is (though it'd be nice to know Travis will pass...).

With that out of the way, I think that tool lints don't need their own section. Instead, we need a more general section on lint check values that says both identifiers and `ident:ident` is valid. We should also probably have a grammar for these attributes, like I added for `cfg_attr` and `cfg` in #444. That we don't say that the lint names must be valid identifiers is an oversight I did not notice until you posted this PR.

**Comment by Manishearth** [comment]
Slightly feel they should be in their own section, since the conditional checking of tool lints is important, prefer to merge as-is.

Added an `ignore` to the test since this is nightly-only right now (stabilized, but hasn't yet reached stable)

File: src/attributes.md [hunk 2]
```diff
@@ -357,6 +357,35 @@ pub mod m3 {
 }
 ```
 
+#### Tool lint attributes (`#[allow(clippy::foo)]`)
```
Comment (Havvy) [inline src/attributes.md]: One nit before I merge: Everything in the parenthesis should not be in the header.

**Comment by Manishearth** [comment]
done
-Manish Goregaokar

**Comment by Havvy** [comment]
💟 Thanks!

**Pull request merged**